[
    {
        "content": "<p>I upgraded my (Ubuntu) dist and since have trouble getting mathlib to build. I'm getting errors like the following:</p>\n<div class=\"codehilite\" data-code-language=\"Lean\"><pre><span></span><code><span class=\"n\">javra2</span><span class=\"bp\">@</span><span class=\"n\">i44pc23</span><span class=\"o\">:</span><span class=\"bp\">/</span><span class=\"n\">data1</span><span class=\"bp\">/</span><span class=\"n\">jakob</span><span class=\"bp\">/</span><span class=\"n\">mathlib</span><span class=\"bp\">$</span> <span class=\"n\">leanproject</span> <span class=\"n\">build</span>\n<span class=\"n\">Building</span> <span class=\"n\">project</span> <span class=\"n\">mathlib</span>\n<span class=\"n\">configuring</span> <span class=\"n\">mathlib</span> <span class=\"mi\">0</span><span class=\"bp\">.</span><span class=\"mi\">1</span>\n<span class=\"bp\">&gt;</span> <span class=\"n\">lean</span> <span class=\"c1\">--make src</span>\n<span class=\"bp\">/</span><span class=\"n\">data1</span><span class=\"bp\">/</span><span class=\"n\">jakob</span><span class=\"bp\">/</span><span class=\"n\">mathlib</span><span class=\"bp\">/</span><span class=\"n\">src</span><span class=\"bp\">/</span><span class=\"n\">topology</span><span class=\"bp\">/</span><span class=\"n\">sheaves</span><span class=\"bp\">/</span><span class=\"n\">sheaf.lean</span><span class=\"o\">:</span><span class=\"mi\">1</span><span class=\"o\">:</span><span class=\"mi\">0</span><span class=\"o\">:</span> <span class=\"n\">error</span><span class=\"o\">:</span> <span class=\"n\">file</span> <span class=\"bp\">'</span><span class=\"n\">category_theory</span><span class=\"bp\">/</span><span class=\"n\">limits</span><span class=\"bp\">/</span><span class=\"n\">punit'</span> <span class=\"n\">not</span> <span class=\"n\">found</span> <span class=\"k\">in</span> <span class=\"n\">the</span> <span class=\"n\">search</span> <span class=\"n\">path</span>\n<span class=\"n\">Use</span> <span class=\"bp\">'</span><span class=\"n\">lean</span> <span class=\"c1\">--path' to see where lean is looking, or https://leanprover-community.github.io/file-not-found.html for more</span>\n</code></pre></div>\n<p>I reckoned this was due to some mistake by mathlibtools and tried to reinstall mathlibtools with <code>pipx reinstall mathlibtools</code> but I'm getting a </p>\n<div class=\"codehilite\" data-code-language=\"Lean\"><pre><span></span><code><span class=\"bp\">/</span><span class=\"n\">home</span><span class=\"bp\">/</span><span class=\"n\">javra2</span><span class=\"bp\">/.</span><span class=\"kn\">local</span><span class=\"bp\">/</span><span class=\"n\">pipx</span><span class=\"bp\">/</span><span class=\"n\">venvs</span><span class=\"bp\">/</span><span class=\"n\">mathlibtools</span><span class=\"bp\">/</span><span class=\"n\">bin</span><span class=\"bp\">/</span><span class=\"n\">python</span><span class=\"o\">:</span> <span class=\"n\">No</span> <span class=\"n\">module</span> <span class=\"n\">named</span> <span class=\"n\">pip</span>\n</code></pre></div>\n<p>which is weird. I do have <code>python3-pip</code> installed.</p>\n<p>Since I've no idea about python, can somebody help?</p>",
        "id": 285363012,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654678711
    },
    {
        "content": "<p><span class=\"user-mention silent\" data-user-id=\"110789\">Jakob von Raumer</span> <a href=\"#narrow/stream/116395-maths/topic/trouble.20setting.20up.20mathlib.20after.20upgrade/near/285363012\">said</a>:</p>\n<blockquote>\n<p>I reckoned this was due to some mistake by mathlibtools and tried to reinstall mathlibtools with <code>pipx reinstall mathlibtools</code> but I'm getting a </p>\n<p><div class=\"codehilite\" data-code-language=\"Lean\"><pre><span></span><code><span class=\"bp\">/</span><span class=\"n\">home</span><span class=\"bp\">/</span><span class=\"n\">javra2</span><span class=\"bp\">/.</span><span class=\"kn\">local</span><span class=\"bp\">/</span><span class=\"n\">pipx</span><span class=\"bp\">/</span><span class=\"n\">venvs</span><span class=\"bp\">/</span><span class=\"n\">mathlibtools</span><span class=\"bp\">/</span><span class=\"n\">bin</span><span class=\"bp\">/</span><span class=\"n\">python</span><span class=\"o\">:</span> <span class=\"n\">No</span> <span class=\"n\">module</span> <span class=\"n\">named</span> <span class=\"n\">pip</span>\n</code></pre></div><br>\n</p>\n</blockquote>\n<p>I've seen this error before when I updated Python. Assuming Ubuntu puts things in the typical place, maybe you can try <code>/usr/bin/python3 -m pip install pipx</code> to reinstall pipx for the new Python version.</p>",
        "id": 285363362,
        "sender_full_name": "Anne Baanen",
        "timestamp": 1654678897
    },
    {
        "content": "<p>\"not found in the search path\" usually means you need to run <code>leanpkg configure</code></p>",
        "id": 285363385,
        "sender_full_name": "Eric Wieser",
        "timestamp": 1654678916
    },
    {
        "content": "<p><span class=\"user-mention silent\" data-user-id=\"238446\">Anne Baanen</span> <a href=\"#narrow/stream/116395-maths/topic/trouble.20setting.20up.20mathlib.20after.20upgrade/near/285363362\">said</a>:</p>\n<blockquote>\n<p>I've seen this error before when I updated Python. Assuming Ubuntu puts things in the typical place, maybe you can try <code>/usr/bin/python3 -m pip install pipx</code> to reinstall pipx for the new Python version.</p>\n</blockquote>\n<p><del>Ahh, that seems to have worked, thanks!</del></p>",
        "id": 285363808,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654679162
    },
    {
        "content": "<p><span class=\"user-mention silent\" data-user-id=\"310045\">Eric Wieser</span> <a href=\"#narrow/stream/116395-maths/topic/trouble.20setting.20up.20mathlib.20after.20upgrade/near/285363385\">said</a>:</p>\n<blockquote>\n<p>\"not found in the search path\" usually means you need to run <code>leanpkg configure</code></p>\n</blockquote>\n<p>Had tried that already, but there's something else wrong I guess</p>",
        "id": 285363868,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654679179
    },
    {
        "content": "<p>Still getting <code>/home/javra2/.local/pipx/venvs/mathlibtools/bin/python: No module named pip</code> :(</p>",
        "id": 285364138,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654679330
    },
    {
        "content": "<p>Seems like something is wrong with some \"virtual environment\" used by pip?</p>",
        "id": 285364729,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654679690
    },
    {
        "content": "<p>Maybe the two problems are actually unrelated to each other, I don't know... But i can't seem to solve either one</p>",
        "id": 285365329,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654680033
    },
    {
        "content": "<p>Okay, the one problem seems to be that <code>category_theory.limits.punit</code> seems to be on the end of my branch (<code>preserves-universes</code>) <span aria-label=\"grimacing\" class=\"emoji emoji-1f62c\" role=\"img\" title=\"grimacing\">:grimacing:</span></p>",
        "id": 285366159,
        "sender_full_name": "Jakob von Raumer",
        "timestamp": 1654680501
    }
]